target/linux/brcmiproc/bcm58800/target.mk

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (C) 2018 Jianhui Zhao <jianhuizhao329@gmail.com>
+# Copyright (C) 2019 Samik Gupta <samik.gupta@broadcom.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+SUBTARGET:=bcm58800
```

Comment:
Is there any reason this cannot be a generic arm64 subtarget, so adding Northstar2 support does not require adding a new subtarget?

Reply:
This structure is based on the feedback from @adrianschmutzler to use the samsung target as the model. The top level samsung target is for the s3c24xx family of SoCs; so, here I made the iProc family of SoCs as the top level target. The sub target for samsung is for the s5pv210 SoC; so, here I made the bcm58800 SoC as the sub target. The samsung target specifies the tq210 board in the image/Makefile; so, I specified the ps225 board in the image/Makefile. I'm open to changing it of course, since the goal is to make it the best fit for OpenWRT. So long as the final structure is acceptable to all the reviewers.

Reply:
I made my comments based on the target structures I'm used to and where I was able to give guidance. If @KanjiMonster is aware of a superior approach, I have no objections.

Reply:
The upstream linux arm64 iproc defconfig support enabled both for stingray and Northstar2, so there should be no reason that we cannot have a arm64 subtarget that supports both.

I found a Northstar2 board at work, so I can actually test if the resulting kernel works.

Reply:
So in your proposal, the target would still be brcmiproc but the subtarget would be renamed to arm64. Is that a correct understanding?

Reply:
Correct. And for ARMv7 targets like Helix4 we would add a second subtarget.

Reply:
Okay I understand. Will make the change.

Reply:
One point to note is that there is an existing target (bcm53xx) that is of the iProc family based on its kernel config.

Reply:
Changed subtarget to arm64